File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s)==0:
            return True
        i=0
        j=0
        while i<len(s):
            if j>=len(t):
                return False
            if s[i]==t[j]:
                i+=1
                j+=1
            else:
                j+=1
        if i>=len(s):
            return True
        return False
                
        

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length()==0){
            return true;
        }
        int i = 0;
        int j = 0;
        while(i<s.length()){
            if(j>=t.length()){
                return false;
            }
            if(s.charAt(i)==t.charAt(j)){
                i+=1;
                j+=1;
            }
            else{
                j+=1;
            }
        }
        if(i>=s.length()){
            return true;
        }
        return false;
    }
}